src/main/java/uk/ac/ebi/eva/commons/mongodb/services/VariantWithAnnotationsService.java

```diff
@@ -0,0 +1,163 @@
+import uk.ac.ebi.eva.commons.mongodb.entity.VariantDocument;
+import uk.ac.ebi.eva.commons.mongodb.entity.subdocuments.VariantSourceEntryMongo;
+import uk.ac.ebi.eva.commons.mongodb.filter.VariantEntityRepositoryFilter;
+import uk.ac.ebi.eva.commons.mongodb.repositories.AnnotationRepository;
+import uk.ac.ebi.eva.commons.mongodb.repositories.VariantRepository;
+import uk.ac.ebi.eva.commons.mongodb.repositories.VariantSourceRepository;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class VariantWithAnnotationsService {
+
+    @Autowired
+    private VariantRepository variantRepository;
+
+    @Autowired
+    private AnnotationRepository annotationRepository;
+
+    @Autowired
+    private VariantSourceRepository variantSourceRepository;
+
+    public List<VariantWithSamplesAndAnnotations> findByGenesAndComplexFilters(List<String> geneIds,
+                                                                               List<VariantEntityRepositoryFilter> filters,
+                                                                               List<String> exclude, Pageable pageable) {
+        return convert(variantRepository.findByGenesAndComplexFilters(geneIds, filters, exclude, pageable));
+    }
+
+    private List<VariantWithSamplesAndAnnotations> convert(List<VariantDocument> variantDocuments) {
+        Table<String, String, List<String>> studyFileIdsToSamples = variantSourceRepository.findAndIndexSamples();
+        Map<String, Set<AnnotationDocument>> indexedAnnotations = annotationRepository
+                .findAndIndexAnnotationsOfVariants(variantDocuments);
+
+        return variantDocuments.stream()
+                .map(variant -> convert(variant, studyFileIdsToSamples, indexedAnnotations.get(variant.getId())))
+                .collect(Collectors.toList());
+    }
+
+    private static VariantWithSamplesAndAnnotations convert(VariantDocument variantDocument,
+                                                            Table<String, String, List<String>> sampleNames,
+                                                            Set<AnnotationDocument> annotations) {
+        VariantWithSamplesAndAnnotations variant = new VariantWithSamplesAndAnnotations(
+                variantDocument.getChromosome(),
+                variantDocument.getStart(),
+                variantDocument.getEnd(),
+                variantDocument.getReference(),
+                variantDocument.getAlternate());
+        variant.setIds(variantDocument.getIds());
+        variant.addSourceEntries(convert(variantDocument.getSourceEntries(), sampleNames));
+        variant.setAnnotations(convert(annotations));
+        return variant;
+    }
+
+    private static Set<Annotation> convert(Set<AnnotationDocument> annotations) {
+        return annotations.stream().map(annotation -> new Annotation(annotation)).collect(Collectors.toSet());
```

Comment:
Yes it will work and not complain as long as there is a constructor that accepts a annotationDocument element.